### DISCUSSION TITLE
How to handle &-- SCSS ampersand within Tailwind?

### DISCUSSION THREAD

**Discussion by BearCooder** [tailwindlabs/tailwindcss#11247] [Help] [answered]
Hello!

I still struggle a bit with "more advanced" concepts when it comes to reuse it with Tailwind. In this case I am struggling how to apply the same effect from SCSS ampersand into Tailwind. I figured it should be just straightforward applying the styles in my class=".." but thats not the case as it is not working.

I asked the specific question with the code example and what I tried with Tailwind on [Stackexchange](https://stackoverflow.com/questions/76256604/how-to-rewrite-sass-ampersand-in-tailwind-css) here.

A short snippet of the SCSS here:

```
.slideshow {
  position: absolute;
  top: 0;
  left: 50%;
  width: 100%;
  height: 100%;
  transform: translateX(-50%);

  &--hero {
    z-index: 3;
    left: 0;
    top: 50%;
    // width: 100%;
    height: 85%;
    transform: translateY(-50%) skewY(-10deg);
    transform-origin: center;
    overflow: hidden;
    box-shadow: 0 0 1rem rgba(0, 0, 0, 0.25);
```

Then the initial SCSS in my HTML:
```
<div class="slideshows">
  <div class="slideshow slideshow--hero">
```

After I rewrite with Tailwind - the second div (slideshow slideshow-hero) is not working the way I thought. 
```
 <div className="position-absolute top-0 left-2/4 w-full h-full translate-x-0">
    <div className="z-30 left-0 top-2/4 h-85/1 -translate-y-2/4 -skew-y-12 origin-center shadow-sm">
```

So can anyone please tell me how I should handle the _&--hero_ within Tailwind?

**Comment by wongjn** [comment, marked as answer]
The ampersand isn't really relevant in Tailwind here. You haven't shown the CSS for `slideshows` but for `slideshow slideshow--hero`, it would be:

```scss
.slideshow {
  position: absolute; // absolute
  top: 0; // top-0 (overridden by .slideshow--hero)
  left: 50%; // left-1/2 (overridden by .slideshow--hero)
  width: 100%; // w-full
  height: 100%; // h-full (overridden by .slideshow--hero)
  transform: translateX(-50%); // -translate-x-1/2 (overridden by .slideshow--hero)

  &--hero {
    z-index: 3; // z-[3]
    left: 0; // left-0
    top: 50%; // top-1/2 
    height: 85%; // h-[85%]
    transform: translateY(-50%) skewY(-10deg); // -translate-1/2 -skew-[10deg]
    transform-origin: center; // origin-center (CSS default, no class needed)
    overflow: hidden; // overflow-hidden
    box-shadow: 0 0 1rem rgba(0, 0, 0, 0.25); // shadow-[0_0_1rem_rgba(0,0,0,.25)]
```

So putting it all together:

```html
<div class="slideshows">
  <div class="absolute top-1/2 left-0 z-[3] w-full h-[85%] -translate-y-1/2 -skew-[10deg] overflow-hidden shadow-[0_0_1rem_rgba(0,0,0,.25)]">
```

**Reply by BearCooder** [reply]
Thank you for the quick response. But shouldnt it look this way when I rewrite the slideshow that it looks this way:?

```
<div class="position-absolute top-0 left-1/2 w-full h-full -translate-x-1/2""> //this is my div class="slideshows"
  <div class="absolute top-1/2 left-0 z-[3] w-full h-[85%] -translate-y-1/2 -skew-[10deg] overflow-hidden shadow-[0_0_1rem_rgba(0,0,0,.25)]"> //this is my div class="slideshow slideshow--hero"
```

Or what am I missing or whats wrong in my approach how I rewrite it?

**Reply by wongjn** [reply]
I do not have knowledge of the `slideshows` CSS so I cannot make a judgement on how that should be transposed but `slideshow slideshow--hero` seems to have been transposed properly.

**Reply by BearCooder** [reply]
Oh yes I just didnt noticed that in the div its written with an s at the end slideshows but in the css its slideshow only. Weird. But this explains my confusion and my followup question. Thank you very much again for helping!
